redfish-core/lib/log_services.hpp

```diff
@@ -1475,6 +1528,219 @@ inline void requestRoutesJournalEventLogEntry(App& app)
             });
 }
 
+template <typename Callback>
+void getHiddenPropertyValue(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                            const std::string& entryId, Callback&& callback)
+{
+    auto respHandler = [callback{std::move(callback)},
+                        asyncResp](const boost::system::error_code ec,
+                                   std::variant<bool>& hiddenProperty) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR << "DBUS response error: " << ec;
+            messages::internalError(asyncResp->res);
+            return;
+        }
+        bool* hiddenPropValPtr = std::get_if<bool>(&hiddenProperty);
+        if (hiddenPropValPtr == nullptr)
+        {
+            messages::internalError(asyncResp->res);
+            return;
+        }
+        bool hiddenPropVal = *hiddenPropValPtr;
+        callback(hiddenPropVal);
+    };
+
+    // Get the Hidden Property
+    crow::connections::systemBus->async_method_call(
+        respHandler, "xyz.openbmc_project.Logging",
+        "/xyz/openbmc_project/logging/entry/" + entryId,
+        "org.freedesktop.DBus.Properties", "Get",
+        "org.open_power.Logging.PEL.Entry", "Hidden");
+}
+
+inline void getDBusLogEntryCollection(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    GetManagedObjectsType& resp, eventLogTypes type)
+{
+    nlohmann::json& entriesArray = asyncResp->res.jsonValue["Members"];
+    entriesArray = nlohmann::json::array();
+    for (auto& objectPath : resp)
+    {
+        uint32_t* id = nullptr;
+        std::time_t timestamp{};
+        std::time_t updateTimestamp{};
+        std::string* severity = nullptr;
+        std::string* message = nullptr;
+        std::string* filePath = nullptr;
+        bool resolved = false;
+        bool hiddenPropVal = false;
+        for (auto& interfaceMap : objectPath.second)
+        {
+            if (interfaceMap.first == "xyz.openbmc_project.Logging.Entry")
+            {
+                for (auto& propertyMap : interfaceMap.second)
+                {
+                    if (propertyMap.first == "Id")
+                    {
+                        id = std::get_if<uint32_t>(&propertyMap.second);
+                        if (id == nullptr)
+                        {
+                            messages::internalError(asyncResp->res);
+                            return;
+                        }
+                    }
+                    else if (propertyMap.first == "Timestamp")
+                    {
+                        const uint64_t* millisTimeStamp =
+                            std::get_if<uint64_t>(&propertyMap.second);
+                        if (millisTimeStamp != nullptr)
```

Comment:
I would perfer 

                        if (millisTimeStamp == nullptr)
                        {
                            messages::internalError(asyncResp->res);
                            return;
                        }
                          timestamp =
                                crow::utility::getTimestamp(*millisTimeStamp);
for all of these

Reply:
Done